ggml.c

```diff
@@ -20065,7 +20077,7 @@ static int gguf_get_or_add_key(struct gguf_context * ctx, const char * key) {
 
     const int n_kv = gguf_get_n_kv(ctx);
 
-    ctx->kv = realloc(ctx->kv, (n_kv + 1) * sizeof(struct gguf_kv));
+    ctx->kv = GGML_ALIGNED_REALLOC(ctx->kv, n_kv * sizeof(struct gguf_kv), (n_kv + 1) * sizeof(struct gguf_kv));
```

Comment:
I initially chose to use `GGML_ALIGNED_ALLOC` for the `kv` and `infos` for no good reason but simply because I wanted to keep the pattern of not using `malloc` anywhere in `ggml.c`. Eventually, I started needing to allocate arrays and string which no longer makes sense to have aligned alloc so obviously I broke the pattern anyway.

Therefore, I think it is better to just switch to using `malloc` in `gguf` and avoid the new `GGML_ALIGNED_REALLOC` macro. Sorry about this complication.